src/Akka.Persistence.Linq2Db.Data.Compatibility.Tests/DataCompatibilitySpecBase.cs

```diff
@@ -0,0 +1,250 @@
+﻿// -----------------------------------------------------------------------
+//  <copyright file="DataCompatibilitySpecBase.cs" company="Akka.NET Project">
+//      Copyright (C) 2009-2022 Lightbend Inc. <http://www.lightbend.com>
+//      Copyright (C) 2013-2022 .NET Foundation <https://github.com/akkadotnet/akka.net>
+//  </copyright>
+// -----------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Akka.Actor;
+using Akka.Configuration;
+using Akka.Hosting;
+using Akka.Persistence.Linq2Db.Data.Compatibility.Tests.Internal;
+using Akka.Persistence.Query;
+using Akka.Persistence.Sql.Compat.Common;
+using Akka.Persistence.Sql.Linq2Db;
+using Akka.Persistence.Sql.Linq2Db.Query;
+using Akka.Streams;
+using FluentAssertions;
+using LanguageExt.UnitsOfMeasure;
+using Xunit;
+using Xunit.Abstractions;
+using Xunit.Sdk;
+
+namespace Akka.Persistence.Linq2Db.Data.Compatibility.Tests
+{
+    public abstract class TestSettings
+    {
+        public abstract string JournalTableName { get; }
+        
+        public abstract string JournalMetadataTableName { get; }
+        
+        public abstract string SnapshotTableName { get; }
+        
+        public abstract string ProviderName { get; }
+        
+        public abstract string CompatibilityMode { get; }
+        
+        public virtual string? SchemaName { get; } = null;
+    }
+    
+    public abstract class DataCompatibilitySpecBase<T>: IAsyncLifetime where T: ITestContainer, new()
+    {
+        protected TestCluster? TestCluster { get; private set; }
+        protected ITestContainer Fixture { get; }
+        protected ITestOutputHelper Output { get; }
+
+        protected abstract TestSettings Settings { get; }
+        
+        protected DataCompatibilitySpecBase(ITestOutputHelper output)
+        {
+            Output = output;
+            Fixture = new T();
+        }
+
+        private Config Config()
+        {
+            return (Config) $@"
```

Comment:
All of the pertinent compatibility settings are really from the derived `TestSettings` types - got it